torch/csrc/autograd/autograd.cpp

```diff
@@ -4,6 +4,7 @@
 #include <torch/csrc/autograd/edge.h>
 #include <torch/csrc/autograd/engine.h>
 #include <torch/csrc/autograd/function.h>
+#include <torch/csrc/autograd/functions/basic_ops.h>
```

Comment:
Why the extra include?

Reply:
Oops forgot to remove this, but turns out we might need it now?

Reply:
You don't add anything to this file. So I don't think you need it here no? Maybe new includes in python_engine yes :)

Reply:
I actually did the fix for cpp run_backward entry point originally as well, but when I ported the same tests over it looked like they were passing even without the fix. So I removed the fix, but left this import on accident :P
From your other comments though it looks like we need the fix for the cpp api after all since there is also the requires_grad/non-requires grad case that I didn't consider.